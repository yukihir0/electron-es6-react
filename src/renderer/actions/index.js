export function increment() {
  return { type: "INCREMENT" };
}
 
export function decrement() {
  return { type: "DECREMENT" };
}

export function asyncIncrement() {
  return dispatch => {
    setTimeout(() => {
      dispatch({type: "ASYNC_INCREMENT"});
    }, 1000);
  };
}

export function asyncDecrement() {
  return dispatch => {
    setTimeout(() => {
      dispatch({type: "ASYNC_DECREMENT"});
    }, 1000);
  };
}
