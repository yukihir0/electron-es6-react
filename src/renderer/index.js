import Hello from './components/hello.jsx'
