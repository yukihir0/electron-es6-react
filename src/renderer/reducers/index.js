const initState = {
  value: 0,
  async_value: 0,
  items: []
};

export default (state = initState, action) => {
  switch (action.type) {
    case "INCREMENT":
      return Object.assign({}, state, {
        value: state.value + 1
      });
    case "DECREMENT":
      return Object.assign({}, state, {
          value: state.value - 1
      });
    case "ASYNC_INCREMENT":
      return Object.assign({}, state, {
        async_value: state.async_value + 1
      });
    case "ASYNC_DECREMENT":
      return Object.assign({}, state, {
          async_value: state.async_value - 1
      });
    case "REQUEST_ITEM":
      return Object.assign({}, state, {
        items: action.items
      });
    default:
      return state;
  }
};
