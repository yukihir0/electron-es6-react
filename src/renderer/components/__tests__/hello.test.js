import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Hello from "../hello";

test("should render text", () => {
  const props = {};
  const element = shallow(
    <Hello {...props} />
  );

  expect(element.find("h1").text()).toBe("Hello Electron!!");
});

test("snapshot", () => {
  const props = {};
  const element = renderer.create(<Hello {...props} />).toJSON();

  expect(element).toMatchSnapshot();
});
