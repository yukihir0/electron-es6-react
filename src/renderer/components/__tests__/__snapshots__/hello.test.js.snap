// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1>
        Hello Electron!!
      </h1>
    </div>
  </div>
</div>
`;
